profiles/serializers_test.py

```diff
@@ -304,3 +309,158 @@ def test_delete_employment(self):
+            'profile_url_medium',
+            'profile_url_small',
+            'pretty_printed_student_id',
+            'work_history',
+            'education',
+        }
+        for key in keys:
+            clone = deepcopy(data)
+            clone[key] = None
+            with self.assertRaises(ValidationError) as ex:
+                ProfileFilledOutSerializer(data=clone).is_valid(raise_exception=True)
+            assert ex.exception.args[0] == {key: ['This field may not be null.']}
+
+            field = ProfileFilledOutSerializer(data=clone).fields[key]
+            if isinstance(field, CharField):
+                # test blank string too
+                clone[key] = ""
+                with self.assertRaises(ValidationError) as ex:
+                    ProfileFilledOutSerializer(data=clone).is_valid(raise_exception=True)
+                assert ex.exception.args[0] == {key: ['This field may not be blank.']}
+
+    def test_work_history(self):
+        """
+        All fields are required after a profile is marked filled_out
+        """
+        profile = self.create_profile()
+
+        serializer = ProfileFilledOutSerializer(profile)
+        data = serializer.to_representation(profile)
+        # end date may be null for work history
+        keys = set(EmploymentSerializer.Meta.fields) - {'end_date'}
+        for key in keys:
+            clone = deepcopy(data)
+            clone['work_history'][0][key] = None
+            with self.assertRaises(ValidationError) as ex:
+                ProfileFilledOutSerializer(data=clone).is_valid(raise_exception=True)
+            assert ex.exception.args[0] == {"work_history": [{key: ['This field may not be null.']}]}
+
+            field = ProfileFilledOutSerializer(data=clone).fields['work_history'].child.fields[key]
+            if isinstance(field, CharField):
+                # test blank string too
+                clone['work_history'][0][key] = ""
+                with self.assertRaises(ValidationError) as ex:
+                    ProfileFilledOutSerializer(data=clone).is_valid(raise_exception=True)
+                assert ex.exception.args[0] == {"work_history": [{key: ['This field may not be blank.']}]}
+
+    def test_work_history_end_date(self):
+        """
+        Make sure end_date can be set to null on filled out profiles
+        """
+        profile = self.create_profile()
+
+        serializer = ProfileFilledOutSerializer(profile)
+        data = serializer.to_representation(profile)
+
+        clone = deepcopy(data)
+        clone['work_history'][0]['end_date'] = None
+        ProfileFilledOutSerializer(data=clone).is_valid(raise_exception=True)
+        # No exception raised
+
+    def test_education(self):
+        """
+        All fields are required after a profile is marked filled_out
+        """
+        profile = self.create_profile()
+
+        serializer = ProfileFilledOutSerializer(profile)
+        data = serializer.to_representation(profile)
```

Comment:
every single test case in this class opens with these 3 lines. at the very least, these should be added to setUp